vcr/util.py

```diff
@@ -92,6 +92,21 @@ def composed(incoming):
 def read_body(request):
     if hasattr(request.body, "read"):
         return request.body.read()
+    if hasattr(request.body, "__iter__") and not isinstance(
+        request.body,
+        (bytearray, bytes, dict, list, str),
+    ):
+        body = list(request.body)
+        if body:
+            if isinstance(body[0], str):
+                return "".join(body).encode("utf-8")
+            elif isinstance(body[0], bytes):
+                return b"".join(body)
+            elif isinstance(body[0], int):
+                return bytes(body)
+            else:
+                raise ValueError(f"Body type {type(body[0])} not supported")
+        return None
```

Comment:
- This line seems not to be covered by any test
- I wonder if returning `b""` or `bytes()` would be more adequate. What do you think?

Reply:
added a test, and returned `b""`